src/pages/study-room/hooks/useExitStudyRoom.ts

```diff
@@ -0,0 +1,25 @@
+import { useBlocker } from 'react-router-dom';
+
+import { getLocalStorage } from '@/utils/storage';
+
+import useExitStudyRoomMutation from './useExitStudyRoomMutation';
+
+const useExitStudyRoom = (studyId: number) => {
+  const { mutate } = useExitStudyRoomMutation();
+  const { state, proceed } = useBlocker(
+    ({ currentLocation, nextLocation }) => currentLocation.pathname !== nextLocation.pathname,
+  );
+
+  const exitStudyRoom = () => {
+    if (state !== 'blocked') return;
+
+    const participantId = getLocalStorage('participantId');
+    mutate(
+      { studyRoomId: studyId, participantId: Number(participantId) },
+      { onSuccess: () => proceed() },
```

Comment:
이렇게 표현해줘도 괜찮을 것 같네요!
```suggestion
      { onSuccess:  proceed },
```

Reply:
그리고 onSuccess 콜백을 useExitStudyRoomMutation 훅을 사용하는 쪽에서 정의해주면 코드를 읽을때 mutate가 되었을 때 어떤 일이 일어날지 더욱 명확하게 이해하기 좋을 것 같아요.